File: 250226/2개씩 그룹짓기/group-of-pairs.py
n = int(input())
nums = list(map(int, input().split()))

nums.sort()

max_num = 0
for i in range (n):
    result = nums[i] + nums[2 * n - 1 - i]
    if max_num < result:
        max_num = result

print(max_num)

File: 250226/2개씩 그룹짓기/group-of-pairs.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int N;
int nums[2000];

int main() {
    cin >> N;

    for (int i = 0; i < 2 * N; i++) {
        cin >> nums[i];
    }

    sort(nums, nums + 2 * N);

    int max{ 0 };
    for(int i = 0; i < N; i++){
        int result = nums[i] + nums[2 * N - 1 - i];
        if(max < result){ max = result; }
    }
    cout << max << "\n";
    
    return 0;
}
